src/components/connect-request.js

```diff
@@ -30,93 +46,47 @@ const connectRequest = (mapPropsToConfig, options = {}) => (WrappedComponent) =>
 
         componentDidMount() {
             const configs = mapPropsToConfig(this.props);
-            if (Array.isArray(configs)) {
-                this.requestAsync(configs, false, true);
-            } else {
-                this.requestAsync(this.props, false, true);
-            }
+            this.requestAsync(configs, false, true);
         }
 
         componentDidUpdate(prevProps) {
-            let prevConfigs = mapPropsToConfig(prevProps);
-            let configs = mapPropsToConfig(this.props);
-            if (!Array.isArray(prevConfigs)) {
-                prevConfigs = [prevConfigs];
-            }
-            if (!Array.isArray(configs)) {
-                configs = [configs];
-            }
+            const prevConfigs = toArray(mapPropsToConfig(prevProps)).filter(Boolean);
+            const configs = toArray(mapPropsToConfig(this.props)).filter(Boolean);
 
-            const diffs = this._diffConfigs(prevConfigs, configs);
-            const toCancel = diffs[0];
-            const toRequest = diffs[1];
+            const { cancelKeys, requestKeys } = diffConfigs(prevConfigs, configs);
+            const requestConfigs = prevConfigs.filter((c) => {
+                return requestKeys.includes(getQueryKey(c.url, c.body));
+            });
 
-            if (toCancel.length) {
-                this.cancelPendingRequests(toCancel);
+            if (cancelKeys.length) {
+                this.cancelPendingRequests(cancelKeys);
             }
-            if (toRequest.length) {
-                this.requestAsync(toRequest);
+            if (requestConfigs.length) {
+                this.requestAsync(requestConfigs);
             }
         }
 
         componentWillUnmount() {
-            this.cancelPendingRequests();
+            const cancelKeys = Object.keys(this._pendingRequests);
```

Comment:
I removed some hasOwnProperty checks and replaced it with a call to Object.keys which seems to do the same thing. Please let me know if there are cases I am not aware of.